packages/neon-api/src/transaction/validator.ts

```diff
@@ -0,0 +1,229 @@
+import { tx, rpc, u } from "@cityofzion/neon-core";
+import { getNetworkFee, getScriptHashesFromTxWitnesses } from "./util";
+
+export enum ValidationAttributes {
+  None = 0,
+  ValidUntilBlock = 1 << 0,
+  SystemFee = 1 << 1,
+  NetworkFee = 1 << 2,
+  Scripts = 1 << 3
+}
+
+export interface ValidationSuggestion<T> {
+  /**
+   * Whether this is auto-fixed by validator
+   * Validator will try to fix the transaction when param `autoFix` is set to TRUE
+   */
+  fixed: boolean;
+  prev: T;
+  suggestion: T;
+}
+
+export interface ValidationResult {
```

Comment:
Consider 
```
{ 
  valid: boolean,
  suggestions: {[prop: string] : ValidationSuggestion<T>
} where prop in Transaction
```

If there is no way to suggest a fix, leave suggestion `null` but offer a human readable message on the issue.

Reply:
Cannot use `[prop: string] : ValidationSuggestion<T>` cause we need to pre-define `T`.
now it's 
```ts
suggestions?: {
    validaUntilBlock?: ValidationSuggestion<number>;
    systemFee?: ValidationSuggestion<u.Fixed8>;
    networkFee?: ValidationSuggestion<u.Fixed8>;
    witnesses?: string;
  };
```

Reply:
Can we have the suggestion reflect the properties of Transaction? So instead of `witnesses` its `scripts`.